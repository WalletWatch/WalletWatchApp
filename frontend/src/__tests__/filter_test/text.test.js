import React from 'react';
import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TextFilter from '../../components/aggrid_filter/filters/text_filter';


describe('TextFilter', () => {
    let mockApi;

    beforeEach(() => {
        const mockDataArray = [
            { data: {'node': "wallet"}, displayed: true },
            { data: {'node': "test"}, displayed: true },
            { data: {'node': "wallet-test"}, displayed: true },
        ];

        mockApi = {
            forEachNode: jest.fn(callback => {
                mockDataArray.forEach(data => callback(data));
            }),
            setColumnFilterModel: jest.fn().mockResolvedValue(),
            onFilterChanged: jest.fn(),
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            isDestroyed: jest.fn().mockReturnValue(false)
        };
    });

    it('renders correctly', () => {
        const { container } = render(
            <TextFilter api={mockApi} />
        );

        expect(container).toMatchSnapshot();
    });

    it('handles checkbox click correctly', () => {
        render(<TextFilter api={mockApi} />);

        const allCheckboxes = screen.getAllByRole('checkbox');
        allCheckboxes.forEach((checkbox) => {
            expect(checkbox).toBeChecked(); 
        });

        const allColsCheckbox = screen.getByTestId('AllParams');
        act(() => {
            userEvent.click(allColsCheckbox);
        });

        allCheckboxes.forEach((checkbox) => {
            expect(checkbox).not.toBeChecked(); 
        });
    });

    it('handle click apply correctly', () => {
        render(<TextFilter api={mockApi} />);

        const allCheckboxes = screen.getAllByRole('checkbox');
        act(() => {
            userEvent.click(allCheckboxes[1]);
        });
        
        act(() => {
            userEvent.click(screen.getByTestId('Apply'))
        });

        setTimeout(() => {
            expect(mockApi.setColumnFilterModel).toHaveBeenCalledWith('node', { values: ["test", "wallet-test"]});
        }, 1000);
    });

    it('handle click apply correctly', () => {
        render(<TextFilter api={mockApi} node/>);

        const allCheckboxes = screen.getAllByRole('checkbox');
        act(() => {
            userEvent.click(allCheckboxes[1]);
        });
        act(() => {
            userEvent.click(screen.getByTestId('Reset'))
        });
        setTimeout(() => {
            expect(allCheckboxes[1]).toBeChecked(); 
        }, 1000);
    });

    it('handle click search correctly', () => {
        render(<TextFilter api={mockApi} node='node'/>);

        const search = screen.getByPlaceholderText('Search for node...');
        act(() => {
            userEvent.type(search, 'test');
        });

        const allCheckboxes = screen.getAllByRole('checkbox');
        expect(allCheckboxes.length).toBe(2);
    });
});



