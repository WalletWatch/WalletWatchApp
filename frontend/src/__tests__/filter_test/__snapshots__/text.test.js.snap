// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFilter renders correctly 1`] = `
<div>
  <div
    class="column-filter"
  >
    <div
      style="overflow: hidden; height: 150px;"
    >
      <input
        class="title"
        id="searcherundefined"
        placeholder="Search for undefined..."
        type="text"
        value=""
      />
      <div
        class="column-filter-wrapper"
        id="searchundefined"
        style="height: calc(100% - 28px);"
      >
        <div
          class="column-filter-item"
        >
          <input
            checked=""
            class="input-cols-filter"
            data-testid="AllParams"
            id="AllParamsundefined"
            type="checkbox"
            value="true"
          />
          <label
            class="input-column-name"
            for="AllParamsundefined"
          >
            (All)
          </label>
        </div>
        <div
          class="column-filter-item"
        >
          <input
            checked=""
            class="input-cols-filter"
            type="checkbox"
            value=""
          />
          <label
            class="input-column-name"
          />
        </div>
      </div>
    </div>
    <div
      class="apply-button-wrapper"
    >
      <button
        class="apply-button"
        data-testid="Reset"
      >
        Reset
      </button>
      <button
        class="apply-button"
        data-testid="Apply"
        style="margin-left: 5px;"
      >
        Apply
      </button>
    </div>
  </div>
</div>
`;
